### PR TITLE
[WIP] Adds a 'closed' property to orgs (so they can be closed w/o deletion)

### PR DESCRIPTION
**Pull request by rossjones**
As described in https://github.com/ckan/ideas-and-roadmap/issues/118

Although it is possible to delete organisations, there is no functionality to allow for organisations to be closed. Closed organisations are those that cease to exist, but whose datasets and other metadata are still important.

Sometimes the organisation's datasets are merged with another, sometimes they're just needed for a historical archive, but deleting them (and thereby removing from public view) is not what is required.
- Allow for marking an organisation as closed (or re-opening).
- Prominently show closed organisations as closed (optionally with a link to the new organisation it may have become part of).
- Ensure that whenever the organisation is shown, the title is rendered with _("(Closed)") appended, or some other way of ensuring that the title clearly shows the organisation no longer exists.

This code contains a nasty workaround to an issue with FactoryBoy that will need to be resolved in model_dictize

### CODE REVIEW

File: ckan/model/group.py [hunk 1]
```diff
@@ -122,17 +125,20 @@ class Group(vdm.sqlalchemy.RevisionedObjectMixin,
...
 
     @property
     def display_name(self):
-        if self.title is not None and len(self.title):
+        if self.title:
+            if self.closed:
+                return "%s (%s)" % (self.title, _("Closed"),)
             return self.title
         else:
             return self.name
```
Comment (davidread) [inline ckan/model/group.py]: I guess we should add "(Closed)" onto this line too?

**Comment by rossjones** [comment]
Still todo:

- [ ] Show organization title default label in autocomplete dropdown for related organization
- [ ] Handle date of closure
- [ ] Find a better way to handle the reasons for the closure.
- [ ] Tests for UI
- [ ] Hide the closed-specific fields in edit-form based on state of closed checkbox.

**Comment by rossjones** [comment]
@amercader  Does that mean thanks but no thanks? :cry:

**Comment by davidread** [comment]
@amercader did you mean to close this?

**Comment by wardi** [comment]
I assume @amercader is taking a well-deserved break, hopefully some place far from a computer. Let me just reopen this so he doesn't have to.

**Comment by amercader** [comment]
Apologies, I was doing a branch clean-up to debug an exception on Read The Docs, and even though I tried to only delete old ones (trac based), I obviously failed.

**Comment by rossjones** [comment]
@amercader totally understandable, no worries.

**Comment by rossjones** [comment]
I think this PR can be safely closed now.  It's unlikely to be rebased, and it could arguably be better implemented with extras. Also, the code's kinda horrible.

**Pull request closed without merging**